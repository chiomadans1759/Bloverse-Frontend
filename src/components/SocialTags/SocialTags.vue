<template>
  <main id="social-tags">
    <div class="item-content-block tags">
      <div class="card" v-for="(link, index) in socialLinks" :key="index">
        <div class="card-content p-1 d-flex justify-content-center">
           <span v-html="link"></span>
           <span class="closeIcon" @click.prevent="removeItem(index)">
             <i class="fal fa-times d-flex justify-content-end close"></i>
            </span>
        </div>
      </div>
  </div>
  </main>
</template>

<script>
export default {
  data(){
    return{
      socialLinks:[
        "<i class='fab fa-facebook-square blue'></i><p>Facebook</p>",
        "<i class='fab fa-twitter sky'></i><p>Twitter</p>",
        "<i class='fab fa-linkedin din'></i><p>LinkedIn</p>"
      ]
    }
  },
  methods: {
    removeItem(index) {
      this.socialLinks.splice(index,1)
      this.$emit('removed')
    },
    reassignIcons(arr){
      const newarr = JSON.stringify(arr);
      this.socialLinks = JSON.parse(newarr);
    }
  }
}
</script>

<style>
#social-tags .blue{
  color: #3B5998;
  margin-top: 0.1rem;
}
#social-tags .din {
  color: #0077B5;
  margin-top: 0.1rem;
}
#social-tags .sky {
  color: #38A1F3;
  margin-top: 0.1rem;
}
#social-tags p {
  display: inline-block;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}
</style>
<style scoped>

#social-tags .card {
  width: 7.031rem;
  height: 1.938rem;
}

#social-tags .tags {
  display: flex;
}

#social-tags .svg-inline--fa {
  cursor: pointer;
  font-size: 0.8rem;
  margin-top: 0.125rem;
}
</style>
