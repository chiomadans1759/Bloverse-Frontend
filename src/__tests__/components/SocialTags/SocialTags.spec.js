import { mount, RouterLinkStub } from '@vue/test-utils';
import SocialTags from '@/components/SocialTags/SocialTags.vue';

describe('SocialTags Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(SocialTags, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should close socialTag', () => {
    const wrapper = mount(SocialTags, {
      stubs: { RouterLink: RouterLinkStub }
    });
    wrapper.find('.remove-icn').trigger('click')
  }),

  it('should reload socialTags', () => {
    const wrapper = mount(SocialTags, {
      stubs: { RouterLink: RouterLinkStub}
    });

    expect(wrapper.vm.reloadSocial).toBe(false);
  })
})
