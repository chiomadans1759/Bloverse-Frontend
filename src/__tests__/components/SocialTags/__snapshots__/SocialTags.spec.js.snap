// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialTags Component should render correctly 1`] = `
<main
  id="social-tags"
>
  <div
    class="item-content-block tags"
  >
    <div
      class="card"
    >
      <div
        class="card-content p-1 d-flex justify-content-center"
      >
        <span>
          <i
            class="fab fa-facebook-square blue"
          />
          <p>
            Facebook
          </p>
        </span>
         
        <span
          class="closeIcon"
        >
          <i
            class="fal fa-times d-flex justify-content-end close"
          />
        </span>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="card-content p-1 d-flex justify-content-center"
      >
        <span>
          <i
            class="fab fa-twitter sky"
          />
          <p>
            Twitter
          </p>
        </span>
         
        <span
          class="closeIcon"
        >
          <i
            class="fal fa-times d-flex justify-content-end close"
          />
        </span>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="card-content p-1 d-flex justify-content-center"
      >
        <span>
          <i
            class="fab fa-linkedin din"
          />
          <p>
            LinkedIn
          </p>
        </span>
         
        <span
          class="closeIcon"
        >
          <i
            class="fal fa-times d-flex justify-content-end close"
          />
        </span>
      </div>
    </div>
  </div>
</main>
`;
